src/App.tsx

```diff
@@ -17,18 +65,46 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                searchQuery={searchQuery}
+                setSearchQuery={setSearchQuery}
+                selectedOption={selectedOption}
+                setSelectedOption={setSelectedOption}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {/* eslint-disable-next-line no-nested-ternary */}
+              {isLoading ? (
+                <Loader />
+              ) : (
+                // create a new component called LoadingError
+                // if the request fails, render the LoadingError component
+                // that will display a message and a button to retry
+                hasLoadingError ? (
+                  <LoadingError
+                    loading={isLoading}
+                    loadTodos={() => fetchTodos}
+                  />
+                ) : (
```

Comment:
Good idea. But let's improve this code

You can move it to `TodoList` and avoid nested ternary